component/core/src/main/java/org/exoplatform/social/core/metadata/attachment/AttachmentServiceImpl.java

```diff
@@ -353,4 +407,54 @@ private FileInfo getAttachment(String objectType, String objectId, String fileId
     return fileInfo;
   }
 
+  private void deleteAttachment(String objectType,
+                                String objectId,
+                                String fileId) throws ObjectNotFoundException, IllegalAccessException {
+
+    List<MetadataItem> metadataItemToDelete =
+                                            metadataService.getMetadataItemsByMetadataNameAndTypeAndObject(fileId,
+                                                                                                           AttachmentService.METADATA_TYPE.getName(),
+                                                                                                           objectType,
+                                                                                                           objectId,
+                                                                                                           0,
+                                                                                                           0);
+    if (!metadataItemToDelete.isEmpty()) {
+      MetadataItem attachmentMetadataItemToDelete = metadataItemToDelete.get(0);
+      metadataService.deleteMetadataItem(attachmentMetadataItemToDelete.getId(), true);
+      broadcastAttachmentDeleted(attachmentMetadataItemToDelete, 0l);
+    }
+  }
+
+  private ObjectAttachmentOperationReport attachUploadFiles(List<String> uploadIds,
+                                                            String objectType,
+                                                            String objectId,
+                                                            String parentObjectId,
+                                                            long userIdentityId) {
+    uploadIds = uploadIds.stream().distinct().toList();
+    ObjectAttachmentOperationReport report = new ObjectAttachmentOperationReport();
+    uploadIds.stream().map(uploadId -> {
+      UploadResource uploadedResource = uploadService.getUploadResource(uploadId);
+      if (uploadedResource == null) {
+        LOG.warn("Uploaded resource with id " + uploadId + " wasn't found");
+        report.addError(uploadId, "attachment.uploadIdNotFound");
+      }
+      return uploadedResource;
+    })
+             .filter(Objects::nonNull)
+             .forEach(uploadResource -> createAttachment(uploadResource,
+                                                         objectType,
+                                                         objectId,
+                                                         parentObjectId,
+                                                         userIdentityId,
+                                                         report));
+    return report;
+  }
+
+  private void broadcastAttachmentDeleted(MetadataItem attachmentMetadataItem, long userIdentityId) {
+    try {
+      this.listenerService.broadcast("attachment.deleted", userIdentityId, attachmentMetadataItem);
```

Comment:
Broadcast using Attachment DTO or id (file id) instead of Metadata API which is a detail of implementation used by the current implementation of this service.

```suggestion
      this.listenerService.broadcast("attachment.deleted", userIdentityId, attachmentMetadataItem);
```